components/eamxx/src/diagnostics/zonal_avg.hpp

```diff
@@ -37,11 +27,19 @@ class ZonalAvgDiag : public AtmosphereDiagnostic {
 #ifdef KOKKOS_ENABLE_CUDA
 public:
 #endif
+  void initialize_impl(const RunType /*run_type*/);
   void compute_diagnostic_impl();
+  // Utility to compute the contraction of a field along its column dimension.
+  // This is equivalent to f_out = einsum('i,i...k->...k', weight, f_in).
+  // The implementation is such that:
+  // - all Field objects must be allocated
+  // - the first dimension for field, weight, and lat is for the columns (COL)
+  // - the first dimension for result is for the zonal bins (CMP,"bin")
+  // - field and result must be the same dimension, up to 3
+  static void compute_zonal_sum(const Field &result, const Field &field, const Field &weight,
```

Comment:
Does this need to be a method at all? Can it just be a local function in the cpp file and be done with it?